packages/pwa-kit-react-sdk/src/utils/performance.js

```diff
@@ -0,0 +1,76 @@
+const NAMESPACE = 'pwa-kit-react-sdk:ssr'
+const PERFORMANCE_MEASUREMENTS = {
+    total: `${NAMESPACE}:total`,
+    renderToString: `${NAMESPACE}:render-to-string`,
+    routeMatching: `${NAMESPACE}:route-matching`,
+    loadComponent: `${NAMESPACE}:load-component`,
+    fetchStrategies: `${NAMESPACE}:fetch-stragegies`,
+    reactQueryPrerender: `${NAMESPACE}:fetch-stragegies:react-query:pre-render`,
+    reactQueryUseQuery: `${NAMESPACE}:fetch-stragegies:react-query:use-query`,
+    getProps: `${NAMESPACE}:fetch-stragegies:get-props`
+}
+export const PERFORMANCE_MARKS = {
+    totalStart: `${PERFORMANCE_MEASUREMENTS.total}:start`,
+    totalEnd: `${PERFORMANCE_MEASUREMENTS.total}:end`,
+    renderToStringStart: `${PERFORMANCE_MEASUREMENTS.renderToString}:start`,
+    renderToStringEnd: `${PERFORMANCE_MEASUREMENTS.renderToString}:end`,
+    routeMatchingStart: `${PERFORMANCE_MEASUREMENTS.routeMatching}:start`,
+    routeMatchingEnd: `${PERFORMANCE_MEASUREMENTS.routeMatching}:end`,
+    loadComponentStart: `${PERFORMANCE_MEASUREMENTS.loadComponent}:start`,
+    loadComponentEnd: `${PERFORMANCE_MEASUREMENTS.loadComponent}:end`,
+    fetchStragegiesStart: `${PERFORMANCE_MEASUREMENTS.fetchStrategies}:start`,
+    fetchStragegiesEnd: `${PERFORMANCE_MEASUREMENTS.fetchStrategies}:end`,
+    reactQueryPrerenderStart: `${PERFORMANCE_MEASUREMENTS.reactQueryPrerender}:start`,
+    reactQueryPrerenderEnd: `${PERFORMANCE_MEASUREMENTS.reactQueryPrerender}:end`,
+    reactQueryUseQueryStart: `${PERFORMANCE_MEASUREMENTS.reactQueryUseQuery}:start`,
+    reactQueryUseQueryEnd: `${PERFORMANCE_MEASUREMENTS.reactQueryUseQuery}:end`,
+    getPropsStart: `${PERFORMANCE_MEASUREMENTS.getProps}:start`,
+    getPropsEnd: `${PERFORMANCE_MEASUREMENTS.getProps}:end`
+}
+
+/**
+ * This is a utility function to get the SSR performance metrics.
+ * The function returns an array of performance measurements.
+ * The data is embedded in the HTML response and will be used to analyze the SSR performance.
+ *
+ * @function
+ * @private
+ *
+ * @return {Array}
+ */
+export const getPerformanceMetrics = () => {
```

Comment:
AWS lambda has a maximum of 1MB of header values roughly. Do we know what that might equate to in # of performance marks? Just asking because we are adding a lot of bytes with these name space stuff. We might seek to condense that a little. 

Food for thought. I don't think we really need to act on this comment.